Sources/BraintreePayPalMessaging/BTPayPalMessagingClient.swift

```diff
@@ -0,0 +1,150 @@
+public class BTPayPalMessagingClient: UIView {
+
+    // MARK: - Properties
+
+    public weak var delegate: BTPayPalMessagingDelegate?
+
+    var apiClient: BTAPIClient
+
+    // MARK: - Initializers
+
+    ///  Initializes a PayPal Messaging client.
+    /// - Parameter apiClient: The Braintree API client
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+
+        super.init(frame: .zero)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public Method
+
+    /// Creates a `BTPayPalMessagingClient` toto be displayed to promote offers such as Pay Later and PayPal Credit to customers.
+    /// - Parameter request: an optional `BTPayPalMessagingRequest`
+    public func createView(_ request: BTPayPalMessagingRequest = BTPayPalMessagingRequest()) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.delegate?.onError(self, error: error)
+                return
+            }
+
+            guard let configuration else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.fetchConfigurationFailed)
+                return
+            }
+
+            guard let clientID = configuration.json?["paypal"]["clientId"].asString() else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.payPalClientIDNotFound)
+                return
+            }
+
+            let messageData = PayPalMessageData(
+                clientID: clientID,
+                environment: configuration.environment == "production" ? .live : .sandbox,
+                amount: request.amount,
+                placement: request.placement?.placementRawValue,
+                offerType: request.offerType?.offerTypeRawValue
+            )
+
+            messageData.buyerCountry = request.buyerCountry
+
+            let messageConfig = PayPalMessageConfig(
+                data: messageData,
+                style: PayPalMessageStyle(
+                    logoType: request.logoType.logoTypeRawValue,
+                    color: request.color.messageColorRawValue,
+                    textAlignment: request.textAlignment.textAlignmentRawValue
+                )
+            )
+
+            let messageView = PayPalMessageView(config: messageConfig, stateDelegate: self, eventDelegate: self)
+            messageView.translatesAutoresizingMaskIntoConstraints = false
+            self.addSubview(messageView)
+
+            NSLayoutConstraint.activate([
+                messageView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+                messageView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+                messageView.topAnchor.constraint(equalTo: self.topAnchor),
+                messageView.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+            ])
```

Comment:
Do we need to have `BTPayPalMessagingClient` conform to `UIView`? What about having a method that returns the merchant a `UIView`?

Reply:
If we want to expose this for merchants using SwiftUI, we need to be bound to a `View` type to have access to the corresponding SwiftUI `Representable` (either `UIViewRepresentable` or `UIViewControllerRepresentable` most commonly). If we don't want to support SwiftUI, we can just have a method that returns a `UIView`, but if we do it's necessary.

Reply:
I get that something has to conform to `UIView`, but I think calling that a `Client` is confusing and goes against our patterns for the other modules. Also in the current design, if they don't call `createView()` nothing works, right? I think that is what's tripping me up about the naming here. Could we re-name Client to View?

Another idea - do we need `createView()` to be a separate method call? What about initializing the `BTPayPalMessagingView/Client` with both an apiClient & an optional request.

Reply:
> Could we re-name Client to View?

It was changed from view to client based on this feedback: https://github.com/braintree/braintree_android/pull/847#discussion_r1422591121. I'm not opposed to changing it back to view but we should align across platforms. cc: @sarahkoop since you originally provided this feedback in the Android PR.

> Another idea - do we need createView() to be a separate method call? What about initializing the BTPayPalMessagingView/Client with both an apiClient & an optional request.

We could move to this pattern if that's preferred. I'm not totally sure how this module could change in the future so it feels a bit less restrictive to have it as a method within the class and also more closely aligns with every module that exists today, but we are also listing this as a beta so we _do_ have the flexibility to change things if we don't like it. What do y'all think?

Reply:
qq: will we instruct merchants to display a loader while the messaging view is configured?

Reply:
We can if we think it'd be helpful, but I think it's pretty common to leave undocumented and up to the merchant to handle any loading screen before the view is displayed.

Reply:
I wonder if the PayPal messaging team could show a [content placeholder](https://www.codepicky.com/content-placeholders/) by default. Then merchants wouldn't have to wait to call `addSubview()`.

Reply:
I'm not sure I follow. We aren't rendering HTML/CSS here we are rendering their raw assets from the native SDK. This is also not the merchant demo this is our implementation and we need to add their subview + add constraints to our view so merchants can actually render these views. If we don't do this the view would never display since we wrap their SDK. We don't have to wait to call it nor will merchants as seen in #1160.

Reply:
I think what Steven means is that currently the time in between the merchant calls `createView()` & `didAppear()` gets fired, just a blank box is shown. I ran this on a device with simulated poor network quality too, and there's potential for that state to last a few seconds.

This would be a bigger change - but it might make more sense to only return the `UIView` instance if we know in fact that it's content is ready to be displayed. So like an async function/init that returns a `UIView`, or throws an error.

Reply:
Ah, gotcha - we should raise this to the messaging team. Imo it makes sense to add some sort of loading indicator to their SDK since they are responsible for the network calls being made before rendering the view. They could add network calls on their end that could impact us knowing when something is ready to display or not so makes since for them to handle that logic end to end. What do you think?

Reply:
Back to[ this comment](https://github.com/braintree/braintree_ios/pull/1153/files#r1430229794) regarding naming with the current pattern, I think the `Client` name could work, if it didn't conform to `UIView`. For example, if it had a method within it that returned the `PayPalMessageView` and the merchant set the delegate on that instead.

Having one `Client` conform to UIView that is rendered on screen, and _requires_ a method call (`createView()`) in addition to it's init for it properly function, is different than the rest of our `Cilent`s.

Reply:
Back to your Demo PR, [doing this](https://github.com/braintree/braintree_ios/blob/1fd50d04042da40fa0f06a7bb5ca024de90af2e9/Demo/Application/Features/PayPalMessagingViewController.swift#L22-L30) `view.addSubview(payPalMessagingClient)` reads odd to me

Reply:
> This is a blocker for supporting SwiftUI easily.

Right, we would need to introduce _another_ class type, that does conform to `View`, if we kept the `Client` name and `Client` didn't conform to View.

Reply:
I think we should mostly leave your PR as-is but revert the rename of `BTPayPalMessagingClient` --> `BTPayPalMessagingView`. Maybe Sarah's initial comment was not considering that this class will conform to View, since in that Android PR the implementation isn't shown there?

Also ➕ 1 to your comment re: bringing up the loading/blank state to the messaging team. It's beyond our scope!

Reply:
> I think we should mostly leave your PR as-is but revert the rename of BTPayPalMessagingClient --> BTPayPalMessagingView.

Yeah I think sometimes too seeing it written out in a demo app context helps as well. We also don't have any modules like this currently so figuring out some uncharted territory. Done: https://github.com/braintree/braintree_ios/pull/1153/commits/13c898d482cea8396811efeff73571947d908a73

> Also ➕ 1 to your comment re: bringing up the loading/blank state to the messaging team. It's beyond our scope!

Already raised it in our slack channel with their team. 😄